### ISSUE TITLE
Mermaid - Gantt diagram and Pie chart are not generated when edited in Editor layout

### ISSUE THREAD

**Issue by MathieuHd** [laurent22/joplin#9973]
### Operating system

Windows

### Joplin version

2.13.15

### Desktop version info

Joplin 2.13.15 (dev, win32)

Client ID: 179ed0eb6b05491b9e23e87662a0d6fb
Sync Version: 3
Profile Version: 44
Keychain Supported: Yes

Revision: 7d2c1c0

Webview test: 1.0.0

## Steps to reproduce
1. Set layout to Editor view
2. Create (or edit) a Mermaid Gantt diagram or Pie chart
3. Wait for 1s and switch to Viewer or Split-view layout
4. The space for the chart is here, download and copy button are here, but the diagram is blank

_Nota: other charts work well (tested flowchart, sequence diagram, class diagram, state diagram, er diagram, journey, quadrant chart, requirement diagram, git graph, C4, mindmap, timeline)_

### Editor view (step 2)
![image](https://github.com/laurent22/joplin/assets/113665805/76fc6a7a-e870-4da2-854b-e2f49a5a42e2)

### Blank render (step 4)
![image](https://github.com/laurent22/joplin/assets/113665805/ab45a44e-4e8e-45c4-9100-36670eafab76)

### Split-view workaround
![image](https://github.com/laurent22/joplin/assets/113665805/ffe4da9d-2c71-4161-9230-ca05d34df2e4)

## Workarounds
- Edit the chart in Split view: it will be generated on edition
- Switch to Rich Text editor
- When in Editor view, quickly switch to Viewer layout in the .5s after editing
- Don't edit the chart: once the diagram is generated, it will stay visible until you edit it again - even if you switch layouts and/or edit the rest of the note

### Expected behaviour

Expected behaviour is that graph are generated even when edited in Editor mode.

## Console errors
When editing graph in Editor layout, Mermaid raises a bunch of errors regarding negative width / radius values for drawing shapes. These do not happen when in Split view layout.

![image](https://github.com/laurent22/joplin/assets/113665805/0217f97e-0a4e-433b-9f07-d5dc684d84bf)

**Comment by abansal15** [comment]
hello @MathieuHd @laurent22  could you please assign this issue to me ?

**Comment by personalizedrefrigerator** [comment]
This seems to be fixed in version 2.14.

For reference, I'm testing with a note with the following content:
````
```mermaid
gantt

title A Gantt diagram
	dateFormat YYYY-MM-DD
	section Section
	A task		:a1, 2014-01-01, 30d
	B task		:a1, 2014-01-01, 10d
	section Another
	A task		:a1, 2014-01-01, 30d
	B task		:a1, 2014-01-01, 10d
```
````

**Comment by MathieuHd** [comment]
I just updated to 2.14.17 and can confirm this was fixed.
Thanks a lot!
